ios/brave-ios/Sources/AIChat/Components/Feedback/AIChatFeedbackView.swift

```diff
@@ -0,0 +1,381 @@
+    get { scalePercent }
+    set { scalePercent = newValue }
+  }
+  
+  func effectValue(size: CGSize) -> ProjectionTransform {
+    let projection = ProjectionTransform(
+      CGAffineTransform(scaleX: 1.0, y: scalePercent)
+    )
+    return ProjectionTransform(CATransform3DIdentity)
+      .concatenating(projection)
+  }
+}
+
+private struct DropdownView<ActionView, MenuView>: View where ActionView: View, MenuView: View {
+  @Binding
+  var showMenu: Bool
+  
+  @ViewBuilder
+  let actionView: () -> ActionView
+  
+  @ViewBuilder
+  let menuView: () -> MenuView
+  
+  var body: some View {
+    actionView()
+      .opacity(showMenu ? 0.7 : 1.0)
+      .zIndex(1)
+      .overlay(alignment: .bottom) {
+        Group {
+          if showMenu {
+            menuView()
+              .transition(
+                .modifier(
+                  active: MenuScaleTransition(scalePercent: 0),
+                  identity: MenuScaleTransition(scalePercent: 1)
+                )
+                .combined(with: .opacity)
+              )
+          }
+        }
+        .alignmentGuide(.bottom) { $0[.top] }
+      }
+      .animation(.easeInOut(duration: 0.20), value: showMenu)
+    }
+}
+
+private struct AIChatDropdownButton: View {
+  var action: () -> Void
+  var title: String
+  
+  var body: some View {
+    Button {
+      action()
+    } label: {
+      Text(title)
+        .font(.subheadline)
+        .foregroundStyle(Color(braveSystemName: .textPrimary))
+        .frame(maxWidth: .infinity, alignment: .leading)
+      
+      Image(braveSystemName: "leo.arrow.small-down")
+        .foregroundStyle(Color(braveSystemName: .iconDefault))
+        .padding(.leading)
+    }
+    .padding(12.0)
+    .background {
+      RoundedRectangle(cornerRadius: 8.0, style: .continuous)
+        .stroke(Color(braveSystemName: .iconDefault), lineWidth: 0.5)
+    }
+    .background(Color(braveSystemName: .containerBackground))
+    .clipShape(RoundedRectangle(cornerRadius: 8.0, style: .continuous))
+  }
+}
+
+private struct AIChatDropdownMenu<Item>: View where Item: RawRepresentable, Item.RawValue: StringProtocol, Item: Identifiable {
+  @Binding
+  var selectedIndex: Int
+  var items: [Item]
+  
+  var body: some View {
+    VStack(spacing: 0.0) {
+      ForEach(Array(items.enumerated()), id: \.offset) { index, option in
+        Button {
+          selectedIndex = index
+        } label: {
+          Text(option.rawValue)
+            .font(.subheadline)
+            .foregroundStyle(Color(braveSystemName: .textPrimary))
+            .frame(maxWidth: .infinity, alignment: .leading)
+          
+          if index == selectedIndex {
+            Image(braveSystemName: "leo.check.normal")
+              .foregroundStyle(Color(braveSystemName: .iconDefault))
+          }
+        }
+        .padding()
+        
+        if index != items.count {
+          Color(braveSystemName: .dividerSubtle)
+            .frame(height: 1.0)
+        }
+      }
+    }
```

Comment:
This is all generally not correct and should be fixed

1. You don't want to an items index to be it's ID. If the item is RawRepresentable, then the ID should be its raw value, not the ID
2. You don't want to use an integer to determine selected item you should be using the ID of the item
3. You should add an `accessibilityRepresentation` + `Picker` for accessibility purposes 

Please see `Picker` SwiftUI definition and rebuild from there